error TS1110: Type expected

**I'm submitting a bug report**

* **Library Version:**
1.5.0 and 1.5.1

**Please tell us about your environment:**
* **Operating System:**
OSX 10.x and Linux Ubuntu

* **Node Version:**
10.14.2

* **NPM Version:**
6.6.0

* **JSPM OR Webpack AND Version**
Webpack

* **Browser:**
not relevant, bug is in building process of other library dependent of this one

* **Language:**
TypeScript 3.6.4 (also tested with older versions of TS)

**Current behavior:**
[Since the new typing definition](https://github.com/aurelia/dependency-injection/releases/tag/1.5.0), we encountered issues trying to build plugins and projects that depend on aurelia-dependency-injection. For exemple, trying to build aurelia-ux project, we get the error: 

```
../../node_modules/aurelia-dependency-injection/dist/aurelia-dependency-injection.d.ts(67,5): error TS1110: Type expected.
[3] ../../node_modules/aurelia-dependency-injection/dist/aurelia-dependency-injection.d.ts(67,5): error TS1110: Type expected.
[4] ../../node_modules/aurelia-dependency-injection/dist/aurelia-dependency-injection.d.ts(67,5): error TS1110: Type expected.
```

**Expected/desired behavior:**
No error when building projects that depend on aurelia-dependency-injection

* **What is the motivation / use case for changing the behavior?**
I'm stuck in several projects due to this bug. I can force projects that have direct dependency to stick with 1.4.2 but it becomes complicated when I work on projects that have dependency to aurelia-dependency-injection through other dependencies...

See Aurelia Discourse discussion here: https://discourse.aurelia.io/t/contributing-to-aurelia-ux/3023

--- a/src/resolvers.ts
+++ b/src/resolvers.ts
@@ -42,7 +42,7 @@ export interface Resolver {
   get(container: Container, key: any): any;
 }
 
-export const enum Strategy {
+export enum Strategy {
   instance = 0,
   singleton = 1,
   transient = 2,

--- a/src/resolvers.ts
+++ b/src/resolvers.ts
@@ -71,12 +71,16 @@ export interface StrategyState<
   [Strategy.singleton]: DependencyCtorOrFunctor<TBase, TImpl, TArgs>;
   [Strategy.transient]: DependencyCtorOrFunctor<TBase, TImpl, TArgs>;
   [Strategy.function]: StrategyFunctor<TBase, TImpl, TArgs>;
-  [Strategy.array]: [{
+  /**
+   * For typings purposes, this is done as ({ get: StrategyFunctor } | TImpl)[]
+   * But it should be understood, and used as [{ get: StrategyFunctor }, ...TImp[]]
+   */
+  [Strategy.array]: ({
     get: (
       container: Container,
       key: PrimitiveOrDependencyCtor<TBase, TImpl, TArgs>
     ) => TImpl
-  }, ...TImpl[]];
+  } | TImpl)[];
   [Strategy.alias]: any;
 }
 
@@ -139,7 +143,7 @@ export class StrategyResolver<
       return this.state(container, key, this);
     }
     if (isStrategy<TBase, TImpl, TArgs, Strategy.array>(this.strategy, Strategy.array, this.state)) {
-      return this.state[0].get(container, key);
+      return (this.state[0] as { get: StrategyFunctor<TBase, TImpl, TArgs> }).get(container, key);
     }
     if (isStrategy<TBase, TImpl, TArgs, Strategy.alias>(this.strategy, Strategy.alias, this.state)) {
       return container.get(this.state) as TImpl;

